src/Miki/Services/UserService.cs

```diff
@@ -50,16 +49,27 @@ public async ValueTask<bool> UserIsBanned(long userId)
         }
 
         /// <inheritdoc />
-        public ValueTask SaveAsync()
+        public async ValueTask<bool> UserIsDonatorAsync(long userId)
         {
-            return unitOfWork.CommitAsync();
+            var donatorRecord = await donatorRepository.GetAsync(userId);
+            if (donatorRecord == null)
+            {
+                return false;
+            }
+            return true;
         }
 
+        /// <inheritdoc />
+        public ValueTask UpdateUserAsync(User user)
+            => repository.EditAsync(user);
+
+        /// <inheritdoc />
+        public ValueTask SaveAsync()
+            => unitOfWork.CommitAsync();
+
         /// <inheritdoc />
         public void Dispose()
-        {
-            unitOfWork?.Dispose();
-        }
+            => unitOfWork?.Dispose();
```

Comment:
Unit of work should never be null, avoid using `?` as it should throw an internal exception instead.

Reply:
That's your code tbh.